src/CloudEvents/Serializer/JsonSerializer.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace CloudEvents\Serializer;
+
+use CloudEvents\CloudEventInterface;
+use CloudEvents\Serializer\Exception\UnsupportedVersionException;
+use CloudEvents\V1\CloudEventInterface as V1CloudEventInterface;
+use DateTimeInterface;
+
+use function array_filter;
+use function array_merge;
+
+class JsonSerializer
+{
+    /**
+     * @throws UnsupportedVersionException
+     * @throws \JsonException
+     */
+    public function serialize(
+        CloudEventInterface $cloudEvent,
+        int $serializationOptions = JsonSerializationOptions::NONE
+    ): string {
+        $payload = $this->createPayload($cloudEvent);
+        $payload = array_merge($payload, $this->formatData($cloudEvent->getData()));
+
+        if ($serializationOptions & JsonSerializationOptions::DROP_NULL_VALUES) {
+            $payload = $this->dropNullValues($payload);
+        }
+
+        return json_encode($payload, JSON_THROW_ON_ERROR);
+    }
+
+    /**
+     * Get a JSON-serializable array representation of the CloudEvent.
+     *
+     * @throws UnsupportedVersionException
+     */
+    protected function createPayload(CloudEventInterface $cloudEvent): array
+    {
+        if ($cloudEvent instanceof V1CloudEventInterface) {
+            return [
+                'specversion' => $cloudEvent->getSpecVersion(),
+                'id' => $cloudEvent->getId(),
+                'type' => $cloudEvent->getType(),
+                'datacontenttype' => $cloudEvent->getDataContentType(),
+                'dataschema' => $cloudEvent->getDataSchema(),
+                'subject' => $cloudEvent->getSubject(),
+                'time' => $this->formatTime($cloudEvent->getTime()),
+            ];
+        }
+
+        throw new UnsupportedVersionException('Unsupported CloudEvent spec version.');
+    }
+
+    private function formatTime(?DateTimeInterface $time): ?string
+    {
+        return $time === null ? null : $time->format(DATE_ATOM);
+    }
+
+    private function dropNullValues(array $payload): array
+    {
+        return array_filter(
+            $payload,
+            static function ($value) {
+                return $value !== null;
+            }
+        );
```

Comment:
Please prefer short functions for single line logic.
```suggestion
        return array_filter($payload, fn ($value) => $value !== null));
```